Implement initial symptom screen within submission (EXPOSUREAPP-2485)

Oliver-Zimmerman opened the pull request:
## Description
Implemented the initial symptom question screen of ticket 1605. 

- Created SubmissionSymptomIntroductionFragment
- Created fragment_submission_symptom_into.xml, following provided design specs.
- Created SubmissionSymptomIntroTest, adapting a test created by Apopov for a similar screen. 
- Added required german strings, and empty strings for translation. 

**Note**: Navigation needs to be sorted when marrying the other subtasks. 

Closes ticket EXPOSUREAPP-2485 in Jira.

Oliver-Zimmerman committed:
initial commit of new fragment for symptom screen

Oliver-Zimmerman committed:
Added relevant strings for symptom screen. Started editing xml of initial symptom screen

Oliver-Zimmerman committed:
Addded 3 selection options within symptom screen

Oliver-Zimmerman committed:
button id name change to something more relevant

Oliver-Zimmerman committed:
ID changes to be something more relevant to symptom scenerio.

Oliver-Zimmerman committed:
Implemented symptom initial screen logic. Temporary nav for testing.

Oliver-Zimmerman committed:
CircleCI formatting

jakobmoellerdev committed:
Remove Cache Clearing to reduce CDN load on failing devices (EXPOSUREAPP-2405) (#1108)

* Remove Cache Clearing to reduce CDN load on failing devices. This ideally needs to be accompanied by a way to clear the cache manually and a way to identify more root causes of Transaction Failures.

Signed-off-by: d067928 <jakob.moeller@sap.com>

* Remove Files that failed for Key Retrieval

Only Remove Files from Cache that failed to download, instead of every file. This is accompanied ideally by no rollback in the Key Retrieval. We only delete the cache ref, the file will not be deleted as it is considered not present

Signed-off-by: d067928 <jakob.moeller@sap.com>

* Introduce dedicated QuotaCalculator for Unit Testing

Signed-off-by: d067928 <jakob.moeller@sap.com>

* Refactor QuotaCalculator for LocalData Property Access and Write Tests

Signed-off-by: d067928 <jakob.moeller@sap.com>

* Use Instant on the Device Read since this is not timezone specific

Signed-off-by: d067928 <jakob.moeller@sap.com>

* Add specific state for the quota calculation and dedicated rollback

Signed-off-by: d067928 <jakob.moeller@sap.com>

* PR Comments

Signed-off-by: d067928 <jakob.moeller@sap.com>

Oliver-Zimmerman committed:
Adapted interoperability test provided by apopov

Oliver-Zimmerman committed:
Added accessibilty announcement on resume, fixed navigation for main page

Oliver-Zimmerman committed:
circleCI indentation fix

harambasicluka commented:
Please add Jira ticket in title and add corresponding labels.

harambasicluka committed:
Feature: Update low risk string (EXPOSUREAPP-1971) (#1125)

service-tip-git committed:
[INTERNAL] Translation delivery: commit by LX Lab (#1126)

Change-Id: I89da1ec6f9d4352cac4b3cf91bff73af47b0f6fe

harambasicluka committed:
Fix: Strings - removed ar & ru and updated default strings (#1128)

chris-cwa commented on Corona-Warn-App/src/main/java/de/rki/coronawarnapp/ui/submission/SubmissionSymptomIntroductionFragment.kt:
```diff
@@ -0,0 +1,144 @@
+        setButtonOnClickListener()
+    }
+
+    private fun setButtonOnClickListener() {
```
I suggest we use a ViewModel and let it handle the logic, like discussed in https://teams.microsoft.com/l/entity/com.microsoft.teamspace.tab.wiki/tab::32332847-46cb-4995-a3eb-e183b8c68f28?context=%7B%22subEntityId%22%3A%22%7B%5C%22pageId%5C%22%3A21%2C%5C%22sectionId%5C%22%3A22%2C%5C%22origin%5C%22%3A2%7D%22%2C%22channelId%22%3A%2219%3Ac51bf302889a44e28119b0efc59b8f7d%40thread.tacv2%22%7D&tenantId=42f7676c-f455-423c-82f6-dc2d99791af7

Oliver-Zimmerman replied:
Hi @chris-cwa, refactoring for this has already started after that meeting and is nearly done. I will push it here before EOD for review.

jakobmoellerdev committed:
Throw Exception in case we have a failed entry to ensure Abortion of the Transaction (EXPOSUREAPP-2405) (#1134)

* Throw Exception in case we have a failed entry to ensure Abortion of the Transaction.

Signed-off-by: d067928 <jakob.moeller@sap.com>

* Adjust test to not expect call to clear cache files

Signed-off-by: d067928 <jakob.moeller@sap.com>

Oliver-Zimmerman committed:
Adaption to ViewModel logic. Not yet complete

Adaption to ViewModel logic, attempting to align with implementation done with some interoporibility tasks: reusable layout files, etc.

janetback committed:
Update strings.xml App-Nutzung ab 16 Jahren (EXPOSUREAPP-2559) (#1120)

Co-authored-by: harambasicluka <64483219+harambasicluka@users.noreply.github.com>
Co-authored-by: Jakob Möller <jakob.moeller@sap.com>

Oliver-Zimmerman committed:
Removed 'Button' announcement on risk card (EXPOSUREAPP-1542) (#945)

* removed button announcement entirely

* removed assosciated test

* removed unneccesarry tests

Co-authored-by: Jakob Möller <jakob.moeller@sap.com>
Co-authored-by: harambasicluka <64483219+harambasicluka@users.noreply.github.com>

jakobmoellerdev committed:
Double Timeout Variables for HTTP Requests and Triple the Timeout for the Transaction to ensure less timeouts are received with bad networking. (EXPOSUREAPP-2190) (#1138)

Signed-off-by: d067928 <jakob.moeller@sap.com>

Oliver-Zimmerman committed:
More ViewModel transformation additions

Oliver-Zimmerman committed:
revert nav from test. Circle CI formatting changes

Oliver-Zimmerman committed:
Update SubmissionSymptomIntroductionFragment.kt

Oliver-Zimmerman committed:
Removed old fragment based test

Oliver-Zimmerman committed:
Will implement navigation during marry session.

Oliver-Zimmerman committed:
Update SubmissionViewModel.kt

harambasicluka committed:
added hypen and forced linebreak 1530 (#1143)

chris-cwa committed:
Fixed klint issues (#1144)

chris-cwa commented on Corona-Warn-App/src/main/java/de/rki/coronawarnapp/ui/viewmodel/SubmissionViewModel.kt:
```diff
@@ -141,4 +153,12 @@ class SubmissionViewModel : ViewModel() {
+        symptomRouteToScreen.value = SymptomIntroductionEvent.NavigateToSymptomCalendar
+    }
+
+    fun navigateToPreviousScreen() {
```
this method should be named like 'onPriviousClicked'

chris-cwa commented on Corona-Warn-App/src/main/java/de/rki/coronawarnapp/ui/viewmodel/SubmissionViewModel.kt:
```diff
@@ -141,4 +153,12 @@ class SubmissionViewModel : ViewModel() {
         }
     }
+
+    fun navigateToSymptomCalendar() {
```
this method should be named like 'onNextClicked'

chris-cwa commented on Corona-Warn-App/src/main/java/de/rki/coronawarnapp/ui/viewmodel/SubmissionViewModel.kt:
```diff
@@ -33,6 +36,8 @@ class SubmissionViewModel : ViewModel() {
 
     val scanStatus: LiveData<Event<ScanStatus>> = _scanStatus
 
+    val symptomRouteToScreen: SingleLiveEvent<SymptomIntroductionEvent> = SingleLiveEvent()
```
why not call this "...Event"?

chris-cwa commented on Corona-Warn-App/src/main/java/de/rki/coronawarnapp/ui/submission/SubmissionSymptomIntroductionFragment.kt:
```diff
@@ -0,0 +1,97 @@
+    }
+
+    private fun onChangeCurrentButtonSelected(state: String?) {
+        if (submissionViewModel.currentButtonSelected.value.toString() !== state) {
```
I would prefer if we observe live data rather than directly read from its value

chris-cwa commented on Corona-Warn-App/src/main/java/de/rki/coronawarnapp/ui/submission/SubmissionSymptomIntroductionFragment.kt:
```diff
@@ -0,0 +1,97 @@
+    }
+
+    private fun onClickButtonVerifyHandler() =
+        onChangeCurrentButtonSelected(binding.symptomChoiceSelection.verifyState)
```
call view model method

chris-cwa commented on Corona-Warn-App/src/main/java/de/rki/coronawarnapp/ui/submission/SubmissionSymptomIntroductionFragment.kt:
```diff
@@ -0,0 +1,97 @@
+        onChangeCurrentButtonSelected(binding.symptomChoiceSelection.verifyState)
+
+    private fun onClickButtonApplyHandler() =
+        onChangeCurrentButtonSelected(binding.symptomChoiceSelection.applyState)
```
call view model method

chris-cwa commented on Corona-Warn-App/src/main/java/de/rki/coronawarnapp/ui/submission/SubmissionSymptomIntroductionFragment.kt:
```diff
@@ -0,0 +1,97 @@
+        onChangeCurrentButtonSelected(binding.symptomChoiceSelection.applyState)
+
+    private fun onClickButtonRejectHandler() =
+        onChangeCurrentButtonSelected(binding.symptomChoiceSelection.rejectState)
```
call view model method

chris-cwa committed:
Merge branch 'dev' into feature/implement-initial-symptom-screen

chris-cwa committed:
introduced data types for symptoms

chris-cwa committed:
represent date by long in millis

Oliver-Zimmerman committed:
Naming changes requested

Oliver-Zimmerman committed:
Better event naming

Oliver-Zimmerman committed:
Update fragment_submission_symptom_intro.xml

harambasicluka commented:
Closed as it's included in: https://github.com/corona-warn-app/cwa-app-android/pull/1170
